addons/project_timesheet_holidays/models/resource_calendar_leaves.py

```diff
@@ -151,15 +151,14 @@ def create(self, vals_list):
         return results
 
     def write(self, vals):
-        start_dates = self.mapped('date_from')
-        end_dates = self.mapped('date_to')
         result = super(ResourceCalendarLeaves, self).write(vals)
-        date_from, date_to = vals.get('date_from'), vals.get('date_to')
-        if date_from or date_to:
-            if any(start_date != date_from or end_date != date_to for start_date, end_date in zip(start_dates, end_dates)):
-                timesheets = self.mapped('timesheet_ids')
-                if timesheets:
-                    timesheets.write({'global_leave_id': False})
-                    timesheets.unlink()
-                    self._timesheet_create_lines()
+        date_from, date_to, calendar_id = vals.get('date_from'), vals.get('date_to'), vals.get('calendar_id')
+        if date_from or date_to or 'calendar_id' in vals:
+            modified_records = self.filtered(lambda r: r.date_from != date_from or r.date_to != date_to or r.calendar_id != calendar_id)
```

Comment:
I think that we should check that the value is not None prior to test if the record value is the same. I know the issue was already there before but I think that we should correct this too. What do you think?
Thanks 🙂

Reply:
> the value is not None

I'm not sure to get what you mean, the value of what? and in what conditions could it be none? :)